homeassistant/components/media_player/mpd.py

```diff
@@ -181,13 +183,34 @@ def supported_media_commands(self):
         """Flag of media commands that are supported."""
         return SUPPORT_MPD
 
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self.currentplaylist
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self.playlists
+
+    def select_source(self, source):
+        """Choose a different available playlist and play it."""
+        self.play_media(MEDIA_TYPE_PLAYLIST, source)
+
     def turn_off(self):
         """Service to send the MPD the command to stop playing."""
         self.client.stop()
 
     def turn_on(self):
         """Service to send the MPD the command to start playing."""
         self.client.play()
+        self._update_playlists()
```

Comment:
Move `self._update_playlists()` into the `update` method. If this is too much traffic, use the `@Throttle` decorator on `_update_playlists`. Example here: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/weather/openweathermap.py#L148